packages/nextjs/src/middleware/FronteggProxy.ts

```diff
@@ -9,6 +9,12 @@ export const FronteggProxy = createProxyServer({
   target: process.env['FRONTEGG_BASE_URL'],
   changeOrigin: true,
   selfHandleResponse: true,
+  /**
+   * We set xfwd to true to avoid next-js buggy implementation of x-forwarded-* headers in version 14.0.2
+   * They set the x-forwarded-port header to be 'undefined' in production environment - https://github.com/vercel/next.js/issues/58295
+   * This causes our proxy middleware to fail on every request
+   */
+  xfwd: true,
```

Comment:
Is it working with older versions of next as well?
What if the customers have a port in dev env?

Reply:
it should take the port by its own, check out this change - https://github.com/vercel/next.js/pull/57815#discussion_r1378313718